chrome/content/zotero/xpcom/recognizePDF.js

```diff
@@ -341,6 +407,8 @@ Zotero.RecognizePDF = new function () {
 	 * @return {Promise}
 	 */
 	async function _recognize(item) {
+		if(!item.isAttachment()) return null;
+		
```

Comment:
Well, `getFilePath()` was failing for me when PDF item recognition was triggered when the previous recognition is already in progress. But that should be solved by additional `canRecognize(item)` and `_processingItemID` checks in `this.recognizeItems`. So probably not needed.

Reply:
Yeah, so `canRecognize(item)` fixes that, because if context menu is opened while PDF item is already in progress, until I press "recognize PDF" a new parent is created and selected which is then inserted into recognition queue.

Reply:
I still don't understand. Why would non-attachment items ever be passed to recognizeItems?

Reply:
Because when context menu is created the selected item is PDF attachment, therefore there's an option to recognize it. But after the item is processed, and context menu is still open and the recognition is triggered again, it gets the automatically selected recognized PDF attachment  parent.

https://github.com/zotero/zotero/blob/d899134e7c53a94cbbc62e0252174e9ce71f7666/chrome/content/zotero/zoteroPane.js#L4469

Reply:
Ah, OK, got it.